app/code/Magento/MediaContent/Model/UnassignAsset.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContent\Model;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\MediaContentApi\Api\UnassignAssetInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Used to unassign relation of the media asset to the media content where the media asset is used
+ */
+class UnassignAsset implements UnassignAssetInterface
+{
+    private const MEDIA_CONTENT_ASSET_TABLE_NAME = 'media_content_asset';
+    private const ASSET_ID = 'asset_id';
+    private const TYPE = 'type';
+    private const ENTITY_ID = 'entity_id';
+    private const FIELD = 'field';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * GetAssetsUsedInContent constructor.
+     *
+     * @param ResourceConnection $resourceConnection
+     * @param LoggerInterface $logger
+     */
+    public function __construct(ResourceConnection $resourceConnection, LoggerInterface $logger)
+    {
+        $this->resourceConnection = $resourceConnection;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(int $assetId, string $contentType, string $contentEntityId, string $contentField): void
```

Comment:
Why we require a large number of arguments? Should the $assetId be enough?

Reply:
A Magento object such as a Category can have several properties keeping the same media asset. For example, image and description. Let's assume that the media asset with the id `1231241` is used in category image and category description. In that case, to clearly identify that the asset is used in a category we need to specify the content of usage.  So we need to set an `assetId` as an asset identifier, `contentType` as an object identifier,  `contentEntityId` of the entity and `contentField` which keeps media asset. In that case, we can guaranty that `Media asset used in` will show a proper result.

Reply:
introduced ContentIdentityInterface